### DISCUSSION TITLE
What's the right way to disable an LSP?

### DISCUSSION THREAD

**Discussion by a-torgovitsky** [LazyVim/LazyVim#4998] [Q&A] [answered]
I have `extras.lang.tex` enabled.
I do not want the `texlab` LSP.
I manually uninstalled it via Mason and put this in my config:
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        texlab = {
          mason = false, -- Don't reinstall automatically
        },
      },
    },
  },
```
That does effectively stop Mason from reinstalling `texlab`, so prevents `texlab` from loading.
However, I get the following warning when I load a `tex` file:
```
   Warn  07:03:02 PM notify.warn ...nt_nvimbZYcwL/usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:800: Spawning language server with cmd: `{ "texlab" }` failed. The language server is either not installed, missing from PATH, or not executable.
```
Is there something else I should be doing to completely disable `texlab`?

**Comment by a-torgovitsky** [comment, marked as answer]
I guessed adding `enabled = false` would work...and it did. But I really just guessed. Any hints on where in the documentation/code I could have looked to figure this out without guessing?

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        texlab = {
          mason = false, -- Don't reinstall automatically
          enabled = false, -- Seemed to work
        },
      },
    },
  },
```

**Reply by folke** [reply]
That's an lspconfig option

**Reply by a-torgovitsky** [reply]
Ok, thanks.
I searched through this: https://github.com/neovim/nvim-lspconfig/blob/master/doc/lspconfig.txt but couldn't find `enabled` anywhere. Am I looking in the wrong place?

**Reply by dpetka2001** [reply]
https://github.com/neovim/nvim-lspconfig/blob/3da18150795cd0cf2c0efe3c6fe6e142800c14ef/lua/lspconfig/configs.lua#L11

**Reply by a-torgovitsky** [reply]
Thanks!
